src/Shared/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorBaselineIntegrationTestBase.cs

```diff
@@ -215,6 +215,15 @@ protected void AssertLinePragmas(RazorCodeDocument codeDocument)
                     Assert.True(foundMatchingPragma, $"No line pragma found for code '{content}' at line {classifiedSpan.Span.LineIndex + 1}.");
                 }
             }
+
+            // check that the pragmas in the main document have matching span maps and are enhanced
+            var pragmasInDocument = linePragmas.Where(p => p.FilePath == codeDocument.Source.FilePath).ToArray();
+
+            foreach(var pragma in pragmasInDocument)
+            {
+               Assert.NotNull(pragma.EndCharacterIndex);
+            }
+            Assert.Equal(pragmasInDocument.Length, csharpDocument.SourceMappings.Length);
```

Comment:
Obviously not an issue for our current tests, but for some reason reading this code did give me a horrible horrible thought: How many things break if a user writes their own `#line` directives in a Razor file? :)

Reply:
Huh, that's an... intriguing point. It will presumably break some of the tooling experiences when they start relying on `#line` directives, although everything should at least compile and run fine.

I guess it falls into the same bucket that a lot of razor stuff does, where it's possible to write C# that breaks the abstraction and 'leaks' into the generated file in bad ways. It would be wonderful to catch them all but until we do we just tell the user 'don't do that' heh.